lib/webtask.js

```diff
@@ -221,9 +221,15 @@ Webtask.prototype.createCronJob = function (options, cb) {
  * @returns {Promise} A Promise that will be fulfilled with the result of inspecting the token.
  */
 Webtask.prototype.inspect = function (options, cb) {
-    options.token = this.token;
+    var promise;
 
-    var promise = this.sandbox.inspectToken(options);
+    if (this.token) {
+        options.token = this.token;
+        promise = this.sandbox.inspectToken(options);
+    }
+    else {
```

Comment:
What if the webtask was obtained without `fetch_code` or `decrypt` and these options are passed to `Webtask#inspect()`?

Maybe we could see if the `Webtask` instance has `undefined` for `secrets` and/or `code` and call `Sandbox#getWebtask()` with whatever is missing, if any, to obtain the delta.

Reply:
I have not thought about it. This is a new behavior, we can define what happens here, some options: 
1. Do what you suggest. 
2. Just throw in `inspect` in security v2 (i.e. not supported API).
3. Just return the webtask (which is what happens now). 
4. Return the webtask *iff* the options match what the webtask has, otherwise throw. 

I'd be more inclined to do (2) more than (1), since logically in security v2 the "inspect" operation is really deprecated and logically replaced by "getWebtask".

Reply:
@tjanczuk (2) is the most correct option. It will probably also require the most changes across clients hoping to support it, whereas (1) keeps this totally transparent.

I have *ANOTHER* option:

5. Change `Sandbox.prototype.inspectWebtask()` to delegate directly to `Sandbox.prototype.getWebtask()` when `authMode === v2`. In this scenario, we _COULD_ `throw` in `Webtask.prototype.inspect` for `authMode === v2`.